File: modules/angular2_material/test/test_url_resolver.ts
import {UrlResolver} from 'angular2/src/core/services/url_resolver';

export class TestUrlResolver extends UrlResolver {
  constructor() {
    super();
  }

  resolve(baseUrl: string, url: string): string {
    // The standard UrlResolver looks for "package:" templateUrls in
    // node_modules, however in our repo we host material widgets at the root.
    if (url.startsWith('package:angular2_material/')) {
      return '/base/dist/js/dev/es5/' + url.substring(8);
    }
    return super.resolve(baseUrl, url);
  }
}

File: modules/angular2_material/test/test_url_resolver.dart
library ng_material.test_url_resolver;

import 'package:angular2/src/platform/browser/browser_adapter.dart';
import 'package:angular2/src/compiler/url_resolver.dart';

void commonDemoSetup() {
  BrowserDomAdapter.makeCurrent();
}

class TestUrlResolver extends UrlResolver {
  @override
  String resolve(String baseUrl, String url) {
    const MATERIAL_PKG = 'package:angular2_material/';

    if (url.startsWith(MATERIAL_PKG)) {
      return '/packages/angular2_material/' +
          url.substring(MATERIAL_PKG.length);
    }
    return super.resolve(baseUrl, url);
  }
}
